### PR TITLE
Fix edge case for lambda with no params

### PR DESCRIPTION
**Pull request by egiurleo**
### Motivation
We didn't have a test case for lambdas with no block parameters. Sorbet represents this case with an empty `Args` node, so we can check whether the `parameters` field inside a `PM_BLOCK_PARAMETERS_NODE` is a `nullptr`, and then return early if so.

### Test plan
See included automated tests.

### CODE REVIEW

File: test/prism_regression/lambda.rb [hunk 1]
```diff
@@ -19,3 +19,6 @@ def method_returning_lambda
     -> { 123 }
   end
 end
+
+# Empty lambda parameters
+->() { 1 + 2 }
```
Comment (amomchilov) [inline test/prism_regression/lambda.rb:24]: Interesting that there's a difference between `-> () {}` and `-> {}`

**Review by amomchilov — APPROVED** [review]
Nice find!
